### PR TITLE
0.8 Preview

### PR DESCRIPTION
**Pull request by ssorallen**
0.8-preview-merger is a throw away branch that is based off master. I created it to be able to show 0.8-preview in a pull request.

This is a port of `core-header-panel` from 0.5 to 0.8.

### CODE REVIEW

File: core-header-panel.html [hunk 1]
```diff
@@ -67,211 +67,261 @@
...
-     *
-     * @event scroll
-     */
-
-    publish: {
+  (function() {
+
+    'use strict';
+
+    function classNames(obj) {
```
Comment (cdata) [inline core-header-panel.html]: Too much copy :spaghetti:

File: core-header-panel.html [hunk 2]
```diff
@@ -67,211 +67,261 @@
...
+      for (var key in obj) {
+        if (obj.hasOwnProperty(key) && obj[key]) {
+          classNames.push(key);
+        }
+      }
+
+      return classNames.join(' ');
+    }
+
+    var MODE_CONFIGS = {
```
Comment (cdata) [inline core-header-panel.html]: Man, I wish that we had a nice place to put class-statics.

File: core-header-panel.html [hunk 3]
```diff
@@ -67,211 +67,261 @@
...
+         * accepts a different class for toggling height, e.g. "medium-tall"
+         *
+         * @attribute tallClass
+         * @type string
+         * @default 'tall'
+         */
+        tallClass: String
+      },
+
+      animateDuration: 200,
```
Comment (cdata) [inline core-header-panel.html]: `configure`?

File: core-header-panel.html [hunk 4]
```diff
@@ -67,211 +67,261 @@
...
+      },
+
+      calcMainContainerClass: function(mode) {
+        return classNames({
+          flex: (mode !== 'cover')
+        });
+      },
+
+      ready: function() {
+        this.scrollHandler = this.scroll.bind(this);
```
Comment (cdata) [inline core-header-panel.html]: `configure`?

File: core-header-panel.html [hunk 5]
```diff
@@ -67,211 +67,261 @@
...
+          shadow: false,
+          tallClass: 'tall'
+        };
+      },
+
+      calcDropShadowClass: function(atTop, mode, shadow) {
+        return classNames({
+          hidden: !this.shadow &&
+            (atTop && MODE_CONFIGS.shadowMode[this.mode] ||
+              MODE_CONFIGS.noShadow[this.mode])
```
Comment (cdata) [inline core-header-panel.html]: Can we break out this conditional logic into something a little more verbose and brain-friendly?

File: metadata.html [hunk 6]
```diff
@@ -10,28 +10,28 @@
 <x-meta id="core-header-panel" label="Header Panel" isContainer group="Core">
```
Comment (cdata) [inline metadata.html]: I think we can just remove this file?

File: core-header-panel.html (hunk 1 above)
Comment (ssorallen) [inline core-header-panel.html]: Yes yes yes, agreed! This will have a permanent place soon.

File: bower.json [hunk 7]
```diff
@@ -1,7 +1,20 @@
 {
   "name": "core-header-panel",
   "private": true,
+  "authors": [
+    "The Polymer Authors"
+  ],
   "dependencies": {
-    "polymer": "Polymer/polymer#master"
+    "polymer": "Polymer/polymer#master",
```
Comment (frankiefu) [inline bower.json]: master -> 0.8-preview

File: core-header-panel.html (hunk 4 above)
Comment (ssorallen) [inline core-header-panel.html]: Not sure this kind of logic belongs in `configure`? Need that ES6 `=>`.

File: core-header-panel.html (hunk 5 above)
Comment (ssorallen) [inline core-header-panel.html]: :+1: https://github.com/Polymer/core-header-panel/commit/1380697d269e1c3cc7cb2f89a6f1035930c4d105

**Comment by ssorallen** [comment]
Closing as initial state of 0.8 with notable changes in the near future:

* Extracting `classNames` to a common place for this and other components to use

**Pull request closed without merging**